blocks/model.py

```diff
@@ -155,8 +155,12 @@ def __init__(self, outputs):
         for brick in bricks:
             if brick not in children and brick not in self.top_bricks:
                 self.top_bricks.append(brick)
-        if len(set(b.name for b in self.top_bricks)) < len(self.top_bricks):
-            raise ValueError("top bricks with the same name")
+        # check there is no occurence in top bricks name
+        names = Counter([brick.name for brick in self.top_bricks])
+        repeated_names = [name for name, count in names.items() if count > 1]
+        if repeated_names:
+            raise ValueError("top bricks with the "
+                             "same name : {}".format(repeated_names))
```

Comment:
I think @rizar commented on this already, but it makes more sense to do `"same name: {}"` and since we're nitpicking anyway, `"same name: {}".format(', '.join(repeated_names))` would be even nicer.